storage/innobase/handler/ha_innodb.cc

```diff
@@ -16319,15 +16319,17 @@ innodb_buffer_pool_evict_uncompressed(void)
 			ut_ad(block->page.in_LRU_list);
 
 			mutex_enter(&block->mutex);
-			if (!buf_LRU_free_page(&block->page, false)) {
-				mutex_exit(&block->mutex);
-				all_evicted = false;
-			} else {
-				mutex_exit(&block->mutex);
+			all_evicted = buf_LRU_free_page(&block->page, false);
```

Comment:
No testcase because I'm not sure how to write one: we'd have have at least two unzip_LRU pages with know IDs, DEBUG_SYNC stop here after the first one, do something remove the 2nd one from unzip_LRU - looks extremely laborious. Maybe I just don't know how to write a simpler one. It certainly helps that xtradb_compressed_columns_* stress-test this.